### ISSUE TITLE
Can't escape insert mode

### ISSUE THREAD

**Issue by stemar94** [vimperator/vimperator-labs#622]
##### Issue type:

- Bug report (maybe)
- Feature request (maybe)

##### Version:

```
Vimperator:	3.15.0 (created: 2016/10/13 15:31:18)
Firefox:	Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:49.0) Gecko/20100101 Firefox/49.0
```

##### Description:

For example on the web interface of https://web.whatsapp.com/ I cannot escape the insert mode since after hitting <Esc> it will immediate switch back into insert mode.

##### Expected behavior:

After hitting <Esc> I will be brought back to normal mode without the immediate switch back to insert mode.

##### Steps to reproduce:

1. Open chat window on the web interface of https://web.whatsapp.com/ .
2. Be stuck in insert mode.

**Comment by SammysHP** [comment]
I think this is a "feature" of whatsapp that always forces focus to the textarea.

**Comment by MPW1412** [comment]
Same here. It started a couple of days ago.

Doesn't matter, weather I press Esc or Strg + ], I can't escape the insert mode on web.whatsapp.com.

**Comment by MPW1412** [comment]
This sounds similar to #14 for me. But I don't know, what was done to resolve it.

**Comment by MPW1412** [comment]
This issue still exists in 3.16.

**Comment by SammysHP** [comment]
Probably nothing we can change if whatsapp always grabs the focus. There might be an unfocus event listener that automatically grabs the focus again.

A possible solution might be to add a rule for the whatsapp domain that disables all focus events. IIRC this is possible with autocommands in Vimperator.

**Comment by MPW1412** [comment]
There is a workaround for google.com, and I think that the browser should be capable to interrupt the focus queue.

**Comment by darnir** [comment]
Maybe we can find a way to block that specific javascript function that causes this?

**Comment by SammysHP** [comment]
Yes, as I said: Create an autocommand (`:help :autocmd`) that disables the part of the Whatsapp JavaScript code that grabs the focus. This should be user specific and not in Vimperator's code.
